libs/backend/src/scan/cast_vote_records/build_report_metadata.ts

```diff
@@ -0,0 +1,211 @@
+  YesNoContest,
+} from '@votingworks/types';
+
+function buildWriteInCandidateSelections(
+  contest: CandidateContest
+): CVR.CandidateSelection[] {
+  if (!contest.allowWriteIns) {
+    return [];
+  }
+
+  const writeInSelections: CVR.CandidateSelection[] = [];
+  for (let i = 0; i < contest.seats; i += 1) {
+    writeInSelections.push({
+      '@type': 'CVR.CandidateSelection',
+      '@id': `write-in-${i}`,
+      IsWriteIn: true,
+    });
+  }
+
+  return writeInSelections;
+}
+
+function buildCandidateContest(
+  contest: CandidateContest
+): CVR.CandidateContest {
+  return {
+    '@id': contest.id,
+    '@type': 'CVR.CandidateContest',
+    Name: contest.title,
+    NumberElected: contest.seats,
+    VotesAllowed: contest.seats,
+    ContestSelection: [
+      ...contest.candidates.map(
+        (candidate): CVR.CandidateSelection => ({
+          '@id': candidate.id,
+          '@type': 'CVR.CandidateSelection',
+          CandidateIds: [candidate.id],
+        })
+      ),
+      ...buildWriteInCandidateSelections(contest),
+    ],
+    PrimaryPartyId: contest.partyId,
+  };
+}
+
+function buildBallotMeasureContest(
+  contest: YesNoContest
+): CVR.BallotMeasureContest {
+  return {
+    '@id': contest.id,
+    '@type': 'CVR.BallotMeasureContest',
+    Name: contest.title,
+    ContestSelection: [
+      {
+        '@type': 'CVR.BallotMeasureSelection',
+        '@id': 'yes',
+        Selection: 'yes',
+      },
+      {
+        '@type': 'CVR.BallotMeasureSelection',
+        '@id': 'no',
+        Selection: 'no',
+      },
+    ],
+  };
+}
+
+function buildContest(
+  contest: AnyContest
+): CVR.CandidateContest | CVR.BallotMeasureContest {
+  return contest.type === 'candidate'
+    ? buildCandidateContest(contest)
+    : buildBallotMeasureContest(contest);
+}
+
+function buildElection({
+  election,
+  electionId,
+  electionScopeId,
+}: {
+  election: Election;
+  electionId: string;
+  electionScopeId: string;
+}): CVR.Election {
+  const allCandidates = election.contests
+    .filter((c) => c.type === 'candidate')
+    .map((c) => (c as CandidateContest).candidates)
+    .flat();
+
+  // The VotingWorks election format nests candidate definitions under
+  // each contest, meaning that it's possible a candidate is defined twice
+  // in separate contests. Although unlikely, we deduplicate by ID here
+  const allCandidatesDeduplicated = [
+    ...new Map(allCandidates.map((c) => [c.id, c])).values(),
+  ];
+
+  return {
+    '@type': 'CVR.Election',
+    '@id': electionId,
+    Name: election.title,
+    Candidate: allCandidatesDeduplicated.map((candidate) => {
+      return {
+        '@type': 'CVR.Candidate',
+        '@id': candidate.id,
+        Name: candidate.name,
+        // The VotingWorks election format allows associating candidates
+        // with multiple parties. Although the Ballot Definition CDF allows
+        // something similar, the Cast Vote Records CDF does not.
+        PartyId: candidate.partyIds?.[0],
```

Comment:
What could we assert? We can't really know the right order after the fact, no?

I'm open to taking it out since we're not going to use it (or anything in the cast vote record `Election` object for that matter) but I would just as soon leave it in because it seems important.

Reply:
Yeah... tricky... I think we should discuss with @mattroe.